tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/internal/findandreplace/overlay/FindReplaceOverlayTest.java

```diff
@@ -45,8 +45,8 @@ public class FindReplaceOverlayTest extends FindReplaceUITest<OverlayAccess> {
 	public OverlayAccess openUIFromTextViewer(TextViewer viewer) {
 		Accessor actionAccessor= new Accessor(getFindReplaceAction(), FindReplaceAction.class);
 		actionAccessor.invoke("showOverlayInEditor", null);
-		Accessor overlayAccessor= new Accessor(actionAccessor.get("overlay"), "org.eclipse.ui.internal.findandreplace.overlay.FindReplaceOverlay", getClass().getClassLoader());
-		return new OverlayAccess(getFindReplaceTarget(), overlayAccessor);
+		FindReplaceOverlay overlay= (FindReplaceOverlay) actionAccessor.get("overlay");
```

Comment:
```suggestion
		FindReplaceOverlay overlay= (FindReplaceOverlay) viewer.getControl().getData(FindReplaceOverlay.class.getName()
```

Just as an example on how one can "transport" an object through the data of the underlying control, of course the `FindReplaceOverlay` needs to set the data accordingly once it is installed in a `TextViewer` and remove itself once it is removed. That way it is very easy to check if a given `TextViewer` has a `FindReplaceOverlay` or not (not only for tests).

If this is needed on multiple places one can even has

`static Optional<FindReplaceOverlay> FindReplaceOverlay.of(TextViewer viewer)`

that encapsulates the details on how a `FindReplaceOverlay` know its connection to a `TextViewer`.

Reply:
That looks like a great idea for further reducing reflective access of the tests (in this case related to the `FindReplaceAction` instead of the dialog/overlay itself) and maybe can also improve the `FindReplaceAction` itself because it could relieve the action from tracking the overlay by just attaching it to the target text editor. I would like to follow-up on this in a subsequent PR.